Add HCP account documentation in users page (#68)
hashicorp/terraform-docs-common: website/docs/cloud-docs/users-teams-organizations/users.mdx

To log in with your HCP account, go to the [Sign In to Terraform Cloud](https://app.terraform.io/) page and click **Continue with HCP account**.

### Linked HCP and Terraform Cloud Accounts

Linking an existing Terraform Cloud account to [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account) lets you sign in with your HCP credentials and access your existing Terraform Cloud organizations.
The first time you log in with your HCP credentials, Terraform Cloud searches for existing Terraform Cloud accounts with the same email address. If Terraform Cloud finds an unlinked account, it asks if you want to link that Terraform Cloud account to your HCP account. If there is no existing Terraform Cloud account with that email address, Terraform Cloud creates one and automatically links it to your HCP account. 

> Once you link your HCP and Terraform Cloud accounts, you will no longer be able to sign in with your Terraform Cloud username and password.

Author: But to we think we should keep the [other](https://github.com/hashicorp/terraform-docs-common/pull/68#discussion_r922278871) `HCP Account linking` section too?

> Once you link your HCP and Terraform Cloud accounts, you will no longer be able to sign in with your Terraform Cloud username and password.

Reviewer: Well yes. And the reason why is that we need actual instructions about how to unlink somewhere. My thought is that here we can provide instructions that provide details about unlinking and what happens as a result and then the "Settings" section really just is going through all of the options in the sidebar one by one, right? So it's just a very brief like "hey this section exists and here's what you can do in it". The other thing I was thinking is - youmentioned that the only option RIGHT NOW is unlinking, but is it possible we add more to that "HCP Account Linking" page in the future? If so, I think it makes sense to keep both.

> ## Creating an Account

Author: https://github.com/hashicorp/terraform-docs-common/pull/68#discussion_r921171587

> ## Creating an Account

Reviewer: OH yes, I'm resolving this as I 100% see your point. I'm so sorry for missing that above!

## Creating an Account

To use Terraform Cloud or Enterprise, you must create an account through one of the following methods:

- **Invitation Email:** When a user sends you an invitation to join an existing Terraform Cloud organization, the email includes a sign-up link. After you create an account, you will automatically join that organization and can begin using Terraform Cloud.
- **Sign-Up Page:** Creating an account requires a username, an email address, and a password. For Terraform Cloud, go to [`https://app.terraform.io/public/signup/account`](https://app.terraform.io/public/signup/account). For Terraform Enterprise, go to `https://<TFE HOSTNAME>/public/signup/account`. After you create an account, you will not belong to any organizations. To begin using Terraform Cloud, you can either [create an organization](/cloud-docs/users-teams-organizations/organizations#creating-organizations) or ask an organization owner to send you an invitation email to join their organization.

> -> **Note:** For Terraform Cloud, you may use [your existing HCP account](#log-in-with-your-hashicorp-cloud-platform-account) by clicking **Continue with HCP account** button from the **Sign-Up Page**.

Reviewer: I'm okay with this note here - thank you for flagging it for me! I think this is a case when it makes sense (I just like to make sure we don't overuse notes throughout the docs).  However, i'm recommending a couple of tiny tweaks:

- I want users who may be interested in this method to actually follow the link to the section where we talk about it/the implications, rather than just blindly doing it. So I took out the "how" part. 
- I think we should say again here that this is recommended. 
- Finally - I think we should move this to the top of the section, just under the header. I think it would be good for users to see this note _before_ they make a separate account. This helps prevent them from going through the steps to create a TFC account and not even read down to this section where we recommend logging in with HCP instead. 

```suggestion
-> **Note:** For Terraform Cloud, we recommend [logging in with your HCP account](#log-in-with-your-hashicorp-cloud-platform-account) instead of creating a separate Terraform Cloud account. 
```

> ## Creating an Account

Author: I'm not sure we can do this for multiple reasons:

- Users still can be invited from TFC and **Continue with HCP account** from TFC's **Sign-Up Page** (see screenshot at the bottom);
- HCP account do not apply to TFE, and there is relevant content in this section for TFE. Though even before my changes there's an arguably wrong statement in the beginning of this section: `...To use Terraform Cloud, you must create...`

![image](https://user-images.githubusercontent.com/2106204/178995879-fb931bc4-3031-40b3-8513-79dbd0a30646.png)

> ## Creating an Account

Author: I changed to something different, can you take another look?

> ## Creating an Account

Reviewer: How did I miss this!? I agree with you and I'm resolving this comment. Thank you for explaining and making sure I saw it!

> The recommended method to access Terraform Cloud is with [your HCP account](#continue-with-hcp-account). Alternatively, you may create an account through one of the following methods:

Reviewer: ```suggestion
We recommend logging in to Terraform Cloud with [your HCP account](#log-in-with-hcp-account). Alternatively, you can create an account through one of the following methods:
```

- **Sign-Up Page:** Creating an account requires a username, an email address, and a password. For Terraform Cloud, go to [`https://app.terraform.io/public/signup/account`](https://app.terraform.io/public/signup/account). For Terraform Enterprise, go to `https://<TFE HOSTNAME>/public/signup/account`. After you create an account, you will not belong to any organizations. To begin using Terraform Cloud, you can either [create an organization](/cloud-docs/users-teams-organizations/organizations#creating-organizations) or ask an organization owner to send you an invitation email to join their organization.

### Continue with HCP Account

Accessing Terraform Cloud with [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account) can be done by clicking **Continue with HCP account** button from the **Sign-In** page. 

In the first sign-in, a new Terraform Cloud account will be created and linked to [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account).

> If there is an existing and still unlinked Terraform Cloud account with the same email address from [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account), you'll be able to link those accounts. After this process you won't be able to sign-in using Terraform Cloud's username and password, though you'll be able to unlink your account in **User Settings** page if you choose to.

Reviewer 2: Do we want to add a sentence that some TFC features will be depreciated (MFA) and managed by HCP?

Author: I didn't add it on purpose. That's because there's detailted instructions on how to proceed in TFC's settings page for each one of those features. I feel like the more we add/repeat ourselves here the harder it will be to keep everything up-to-date.

To me the main intention of this change is to warn the user that HCP is a thing (and recommended) and that he won't be able to sign-in with TFC's username/password. I'm not 100% sure about this, that's why I've reinforced feedback request in our channel 😄

> In the first sign-in, a new Terraform Cloud account will be created and linked to [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account).
> 
> If there is an existing and still unlinked Terraform Cloud account with the same email address from [your HCP account](https://cloud.hashicorp.com/docs/hcp/create-account), you'll be able to link those accounts. After this process you won't be able to sign-in using Terraform Cloud's username and password, though you'll be able to unlink your account in **User Settings** page if you choose to.

Reviewer: ```suggestion
### Linking HCP and Terraform Cloud Accounts

You can link an existing Terraform Cloud account to your HCP account so that you can sign in with your HCP credentials and access your existing Terraform Cloud organizations.

The first time you log in with your HCP credentials, Terraform Cloud will search for existing Terraform Cloud accounts with the same email address. If Terraform Cloud finds an unlinked account, it asks you if you want to link that Terraform Cloud account to your HCP account. If there is no existing Terraform Cloud account with that email address, Terraform Cloud creates one and automatically links it to your HCP account. 

Once you link your HCP and Terraform Cloud accounts, you will no longer be able to sign in with your Terraform Cloud username and password.
```

Reviewer: Thank you for being so patient! I don't know what happened (this is why I should be more careful when editing through the GitHub UI. It was a bit repetitive. I think this is how I meant everything to read - just one sentence in "Log in" that leads into "linked accounts" :-) 

## Log In With Your HCP Cloud Account

We recommend using a [HashiCorp Cloud Platform (HCP)](https://portal.cloud.hashicorp.com/sign-up) account to log in to Terraform Cloud. Your HCP Account grants access to every HashiCorp product, including HashiCorp Learn and the Terraform Registry. If you use your HCP Account, you will manage user settings like multi-factor authentication and password resets from within HCP instead of the Terraform Cloud UI. 
	
To log in with your HCP account, go to the [Sign In to Terraform Cloud](https://app.terraform.io/) page and click **Continue with HCP account**. Terraform Cloud may ask if you want to link your HCP and Terraform Cloud accounts.

### Linked HCP and Terraform Cloud Accounts

You must link your HCP Account with a Terraform Cloud account before you can log into Terraform Cloud with your HCP credentials. The first time you click **Continue with HCP account**, Terraform Cloud searches for existing Terraform Cloud accounts with the same email address. If there is an unlinked account, Terraform Cloud asks if you want to link it to your HCP account. If there is no existing Terraform Cloud account with that email address, Terraform Cloud creates one and automatically links it to your HCP account. 
	
You can unlink your Terraform Cloud account from your HCP account if your Terraform Cloud account existed before linking. You cannot unlink an autogenerated Terraform Cloud account from your HCP account.
	
To unlink your HCP account from your Terraform Cloud account:
	
	 1. Go to **User settings** and click **HCP Account Linking** in the sidebar. 
	 1. Click ****Unlink account**. Your Terraform Cloud account is no longer linked to your HCP account, and you can begin using your Terraform Cloud credentials to log in. You cannot log in with your HCP account unless you re-link it to your Terraform Cloud account.

Author: Thanks @laurapacilio! I agree that we're almost there 😄 

About the label button the correct label is **Continue with HCP account**. We're using an old label in staging. There's a [PR](https://github.com/hashicorp/atlas/pull/12881) in the oven to change this label already.

> 

Reviewer: Can we include more detailed instructions about how to unlink? I think something like "Go to Settings and click X" is preferred :)

Refer to [Two-Factor Authentication](/cloud-docs/users-teams-organizations/2fa) for details.

> ### Unlink HCP Account from Terraform Cloud account

Author: Yeah the sidebar is `HCP Account Linking`. Seems a bit confusing but I think the design team intended this menu to convey all available linking options. Unfortunatelly the only current option is unlinking 😬

> ### Unlink HCP Account from Terraform Cloud account

Reviewer: OH okay that makes sense! Okay so it says "HCP Account Linking"? Can we use that? I updated my suggestion to that in case it's helpful!

> Click **HCP Account Linking** in the sidebar to review and unlink [your HCP account](#linked-hcp-and-terraform-cloud-accounts) by clicking the **Unlink account** button.

Reviewer: @andrealbinop I basically want to do the opposite of what you did originally. I think we should have more details about unlinking in the _first_ section that's all about linked accounts and link to that section from this one. This one is a rundown of all the available settings, so I feel like this makes sense. This way, we don't repeat ourselves _too_ much, but we're still providing users with the information in the places they'll expect it!
